src/Microsoft.AspNetCore.WebUtilities/MultipartReaderStream.cs

```diff
@@ -40,15 +38,8 @@ public MultipartReaderStream(BufferedReadStream stream, string boundary, bool ex
 
             _innerStream = stream;
             _innerOffset = _innerStream.CanSeek ? _innerStream.Position : 0;
-            if (expectLeadingCrlf)
-            {
-                _boundaryBytes = Encoding.UTF8.GetBytes("\r\n--" + boundary);
-            }
-            else
-            {
-                _boundaryBytes = Encoding.UTF8.GetBytes("--" + boundary);
-            }
-            _finalBoundaryLength = _boundaryBytes.Length + 2; // Include the final '--' terminator.
+            _boundary = boundary;
+            _boundary.ExpectLeadingCrlf = expectLeadingCrlf;
```

Comment:
I can't understand why is there an optional parameter. But with your change it doesn't make sense to pass it in anymore since it can be set on the `MultipartBoundary` object. 
@Tratcher @davidfowl couldn't we just remove this now and set it at the call site as mentioned.